nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,232 @@
+"""
+Mask nifti images by spherical volumes for seed-region analyses
+"""
+import numpy as np
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn import neighbors
+from sklearn.externals.joblib import Memory
+
+from .. import _utils
+from .._utils import logger, CacheMixin
+from .._utils.niimg_conversions import check_niimg, check_niimg_3d
+from .. import signal
+from .. import image
+from .. import masking
+
+
+def _iter_signals_from_spheres(seeds, niimg, radius, mask_img=None):
+    seeds = list(seeds)
+    niimg = check_niimg(niimg)
+    affine = niimg.get_affine()
+
+    # Compute world coordinates of all in-mask voxels.
+
+    if mask_img is not None:
+        mask_img = check_niimg_3d(mask_img)
+        mask_img = image.resample_img(mask_img, target_affine=affine,
+                                      target_shape=niimg.shape[:3],
+                                      interpolation='nearest')
+        mask, _ = masking._load_mask_img(mask_img)
+        mask_coords = list(np.where(mask != 0))
+
+        X = masking._apply_mask_fmri(niimg, mask_img)
+    else:
+        mask_coords = list(zip(*np.ndindex(niimg.shape[:3])))
+        X = niimg.get_data().reshape([-1, niimg.shape[3]]).T
+    mask_coords.append(np.ones(len(mask_coords[0]), dtype=np.int))
+    mask_coords = np.asarray(mask_coords)
+    mask_coords = np.dot(affine, mask_coords)[:3].T
+
+    if radius is not None:
+        # Fix for early version of sklearn (not sure)
+        radius += 1e-6
+
+    clf = neighbors.NearestNeighbors(radius=radius)
+    A = clf.fit(mask_coords).radius_neighbors_graph(seeds)
+    A = A.tolil()
+    # Include selfs
+    mask_coords = mask_coords.astype(int).tolist()
+    for i, seed in enumerate(seeds):
+        try:
+            A[i, mask_coords.index(list(seeds[i]))] = True
+        except ValueError:
+            # seed is not in the mask
+            pass
+    del mask_coords
+
+    for i, row in enumerate(A.rows):
+        if len(row) == 0:
+            raise ValueError('Sphere around seed #%i is empty' % i)
+        yield X[:, row]
+
+
+def _signals_from_spheres(seeds, niimg, radius, mask_img=None):
+    seeds = list(seeds)
+    n_seeds = len(seeds)
+
+    signals = np.empty((niimg.shape[3], n_seeds))
+    for i, sphere in enumerate(_iter_signals_from_spheres(
+            seeds, niimg, radius, mask_img=mask_img)):
+        signals[:, i] = np.mean(sphere, axis=1)
```

Comment:
Yes! But for further work, for several reasons:
* people will argue about what aggregation functions should be offered by default, wording, doc, etc...
* I don't have an example showing the interest of using another aggregation function.

But I definitely want to do a PR in that sense after this one.